framework/ChiMesh/SweepUtilities/AngleSet/angleset.cc

```diff
@@ -161,12 +161,13 @@ PsiBndry(uint64_t bndry_map,
 {
   if (ref_boundaries[bndry_map]->IsReflecting())
     return ref_boundaries[bndry_map]->HeterogenousPsiIncoming(
-      angle_num, cell_local_id, face_num, fi, gs_ss_begin);
+      cell_local_id, face_num, fi, angle_num, g, gs_ss_begin);
 
   if (not surface_source_active)
-    return &ref_boundaries[bndry_map]->zero_boundary_flux[g];
+    return ref_boundaries[bndry_map]->ZeroFlux(g);
 
-  return &ref_boundaries[bndry_map]->boundary_flux[g];
+  return ref_boundaries[bndry_map]->HeterogenousPsiIncoming(
+    cell_local_id, face_num, fi, angle_num, g, gs_ss_begin);
```

Comment:
I understand we use the `zero_boundary_flux` because we return a pointer to a double, but is there a better way to do this without having a to have zero vector?

Reply:
All ears if you find a better way whilst maintaining the cache behavior within the sweep.